Cannot install datatable from a source distribution

```
...
  /home/pasha/llvm4/bin/clang++ -isystem /usr/include/python3.6m -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Ic -I/usr/include/python3.6m -I/home/pasha/py36-test/include/python3.6m -c c/stats.cc -o build/temp.linux-x86_64-3.6/c/stats.o -fopenmp -isystem /usr/include/python3.6m -I/usr/include/python3.6m -std=gnu++11 -stdlib=libc++ -x c++ -Ic -I/home/pasha/py36-test/include -I/home/pasha/llvm4/include/c++/v1 -I/home/pasha/llvm4/include -isystem /home/pasha/llvm4/include/c++/v1 -O3 -Weverything -Wno-covered-switch-default -Wno-float-equal -Wno-gnu-statement-expression -Wno-switch-enum -Wno-old-style-cast -Wno-c++98-compat-pedantic -Wno-nested-anon-types -Wno-c99-extensions -Wno-deprecated -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Wno-weak-vtables -Wno-weak-template-vtables
  c/stats.cc:8:10: fatal error: 'stats.h' file not found
  #include "stats.h"
           ^~~~~~~~~
  1 error generated.
  error: command '/home/pasha/llvm4/bin/clang++' failed with exit status 1
```

The issue appears to be that `.h` files were not properly included in the source distribution.

--- a/Makefile
+++ b/Makefile
@@ -119,6 +119,9 @@ dist: build
 dist_noomp: build_noomp
 	$(PYTHON) setup.py bdist_wheel -d $(DIST_DIR)
 
+sdist:
+	$(PYTHON) setup.py sdist
+
 version:
 	@$(PYTHON) setup.py --version
 

--- a/setup.py
+++ b/setup.py
@@ -17,8 +17,8 @@ import sys
 import re
 import sysconfig
 from functools import lru_cache as memoize
-from setuptools import setup, find_packages
-from distutils.core import Extension
+from setuptools import setup, find_packages, Extension
+# from distutils.core import Extension
 from sys import stderr
 
 
@@ -47,14 +47,18 @@ def get_version():
     return version
 
 
-def get_c_sources():
-    """Find all C source files in the "c/" directory."""
-    c_sources = []
-    for root, dirs, files in os.walk("c"):
+def get_c_sources(folder, include_headers=False):
+    """Find all C/C++ source files in the `folder` directory."""
+    allowed_extensions = [".c", ".C", ".cc", ".cpp", ".cxx", ".c++"]
+    if include_headers:
+        allowed_extensions.extend([".h", ".hpp"])
+    sources = []
+    for root, dirs, files in os.walk(folder):
         for name in files:
-            if name.endswith(".c") or name.endswith(".cc"):
-                c_sources.append(os.path.join(root, name))
-    return c_sources
+            ext = os.path.splitext(name)[1]
+            if ext in allowed_extensions:
+                sources.append(os.path.join(root, name))
+    return sources
 
 
 def get_py_sources():
@@ -397,9 +401,10 @@ setup(
         Extension(
             "datatable/lib/_datatable",
             include_dirs=["c"],
-            sources=get_c_sources(),
+            sources=get_c_sources("c", include_headers=(argcmd == "sdist")),
             extra_compile_args=get_extra_compile_flags(),
             extra_link_args=get_extra_link_args(),
+            language="c++",
         ),
     ],
 

